catalog/resource_sql_table.go

```diff
@@ -668,13 +684,38 @@ func ResourceSqlTable() common.Resource {
 			if err != nil {
 				return err
 			}
+			partitionInfo, err := NewSqlTablesAPI(ctx, c).getPartitions(d.Id())
+			if err != nil {
+				return err
+			}
+			partitionIndexes := map[int]string{}
 			for i := range ti.ColumnInfos {
 				c := &ti.ColumnInfos[i]
 				c.Identity, err = reconstructIdentity(c)
 				if err != nil {
 					return err
 				}
 			}
+
+			for i := range partitionInfo.ColumnInfos {
+				c := &partitionInfo.ColumnInfos[i]
+				if c.PartitionIndex != nil {
+					partitionIndexes[*c.PartitionIndex] = c.Name
+				}
+			}
+			indexes := []int{}
+			partitions := []string{}
+
+			for index := range partitionIndexes {
+				indexes = append(indexes, index)
+			}
+			sort.Ints(indexes)
+
+			for _, p := range indexes {
+				partitions = append(partitions, partitionIndexes[p])
+			}
```

Comment:
What are the returned partition indices like? Is it always a contiguous set of indices starting from 0?

Reply:
yes, this is the position of the column in the partition (starting from 0), NULL if not a partitioning column.